tests/main/econnreset/task.yaml

```diff
@@ -5,21 +5,25 @@ restore: |
 execute: |
     echo "Downloading a large snap in the background"
     su -c "/usr/bin/env SNAPD_DEBUG=1 snap download --edge test-snapd-huge 2>snap-download.log" test &
-    echo "Wait until it actually got some data for the snap"
-    while ! grep -q "Retrying.*download-snap/.*\.snap, attempt 1" snap-download.log; do
-        sleep 1
+
+    echo "Wait until the download started and downloaded more than 1 MB"
+    partial=test-snapd-huge_1.snap.partial
+    for i in $(seq 20); do
+        [ -f $partial ] && [ $(wc -c <"$partial") -gt 1048576 ] && break
```

Comment:
Additionally, it would be nice to explain briefly why we check for `1048576` exactly, where does that number come from, or express it as `1024 * 1024` and put a note saying that 1mb is enough for us to consider the download as started.